packages/openapi-ts/src/utils/write/types.ts

```diff
@@ -355,10 +360,33 @@ const processServiceTypes = (client: Client, onNode: OnNode) => {
             ...emptyModel,
             export: 'any-of',
             isRequired: true,
-            properties: getSuccessResponses(operation.results),
+            properties: responses,
           }),
         });
 
+        if (config.types.dates === 'types+transform') {
+          if (responses.length === 1) {
+            if (client.types[responses[0].type]?.hasTransformer) {
+              const name = operationResponseTypeName(operation.name);
+              const transformAlias = compiler.transform.alias({
+                existingName: responses[0].type,
+                name,
+              });
+
+              client.types[name].hasTransformer = true;
+
+              onNode(transformAlias);
+            }
+          } else if (responses.length > 1) {
+            console.log(
+              '❗ Route',
+              operation.method,
+              operation.path,
+              'has more than 1 success response and will not currently have a transform generated. If you have a use case for this please open an issue https://github.com/hey-api/openapi-ts/issues',
+            );
+          }
+        }
```

Comment:
Warning added here. I think we can fix this long term by making responseTransformer take `Record<StatusCode, (data: unknown) => unknown>` but it's definitely an uncommon corner case so I'd be happy to shorten the life of this PR and merge it - users can always import the appropriate transform(s) for themselves as they'll still be generated

![image](https://github.com/hey-api/openapi-ts/assets/8896153/020027e3-b7b5-4d63-9752-ed769aeef08b)

Reply:
Yep, let's get it in and see if anyone needs it first